scheduler/src/cook/regexp_tools.clj

```diff
@@ -30,3 +30,9 @@
         (get field-name))
     (catch Exception e
       (throw (ex-info "Failed matching key" {:regexp-name regexp-name :field-name field-name :match-list match-list :key key} e)))))
+
+(defn match-based-on-pool-name
```

Comment:
This move could also be refactored into its own PR.

Reply:
Discussed offline the pros/cons of splitting this up, I decided to keep it as one PR